### ISSUE TITLE
strange problem: app1 gets file upload from app2 -  is it me or a bug?

### ISSUE THREAD

**Issue by benjamin79** [CollectionFS/Meteor-CollectionFS#510]
I´ve two apps which are connected through the same mongodb. When a pdf file is uploaded in app2 localhost:4000, app1 which is on localhost:3000 receives the file too.

I don´t get it why and how to prevent it.

The following code is the same on both apps. The difference is, that in app2 the pdf is also saved as a png file. 
```
 Filesystem = new FS.Collection("pdf", {
   stores: [  new FS.Store.FileSystem("pdf", {
     path: "~/uploads"
   })
   ]
 })

**Comment by mxab** [comment]
Do both collections have the same name ````new FS.Collection("pdf",...```` ? As far as I know the FileWorker runs through all newly added files and copies them in the store, if both collections are named "pdf" I guess the fileworker will try to add it in each of the stores

**Comment by benjamin79** [comment]
yes. I need app2 for uploading and app1 for listing the files. So i think i can´t change pdf to something else?

**Comment by mxab** [comment]
ok but why do you have a png store on the app2, I think that logic is insert a file then copy the binaries it into all it's stores.
Do you generate a png version of the pdf and when yes, why only in app2 and not in the one that actually inserts/uploads the file

**Comment by benjamin79** [comment]
I don´t really get what you mean.

app2 is the one which gets the uploaded file. app1 is only for display. app2 extracts the first found image in the pdf and saves it as a png file + the original one in another store but in the same collection.

**Comment by aldeed** [comment]
What do you mean by "app1 receives the file too"? How does it receive the file? The file will of course appear in both app's collections since they have the same database, but I don't think it will be stored additional times.

**Comment by benjamin79** [comment]
what i expect:
- app2 gets pdf file
- transformWrite saves a png of the pdf to the pdf store in  ~/uploads 
- pdf1 store gets the original pdf file saved in ~/uploads2

what happens:
- everything above 
- ~/uploads receives the original pdf file too.

For testing i changes the path "~/uploads" to "~/uploads3" in app1 and the pdf file shows. So it seems, that the code from app1 is responsible. When the app1 code is removed, everything is ok. But i somehow need to display the uploaded files from app2 in app1.

**Comment by aldeed** [comment]
It's probably because the file worker is not really safe for multiple instances right now. You can try removing `cfs:standard-packages` from app1 and add the component packages individually, leaving out the `cfs:worker` package. The list is [here](https://github.com/CollectionFS/Meteor-CollectionFS/blob/devel/package.js#L13-L26). You may be able to skip adding the upload pkg too if app1 is only for viewing.

@raix, any other ideas?

**Comment by raix** [comment]
@benjamin79 are both apps using same mongodb?

**Comment by benjamin79** [comment]
yes

**Comment by justinventuri** [comment]
I changed FS.FileWorker.observe in cfs_worker.js  to only listen for files that have metadata that match the name we gave our tempstore ie.
```
fsFile = new FS.File()
fsFile.metadata.tempStoreName = "your-tempstore-name"
```
and then
```
if ( FS.TempStore.Storage.name  &&   fsFile.metadata.tempStoreName ){...FS.TempStore.removeFile(fsFile)}
```

**Comment by benjamin79** [comment]
@raix @aldeed 
any solution coming up? When i remove cfs:worker it doesn´t save "copies" with the filename in the filesystem into the document. So how would i download a file without the worker?

And i see DEPRECATING-cfs-worker -> core also.

 @justinventuri  thanks for your idea

**Comment by boomfly** [comment]
+1 
same issue
